src/main/webapp/app/exercises/modeling/assess/modeling-assessment.component.ts

```diff
@@ -141,7 +140,7 @@ export class ModelingAssessmentComponent extends ModelingComponent implements Af
      * element feedback mapping.
      * Returns an array containing all feedback entries from the mapping.
      */
-    private generateFeedbackFromAssessment(assessments: Assessment[]): Feedback[] {
+    generateFeedbackFromAssessment(assessments: Assessment[]): Feedback[] {
```

Comment:
I agree, this was a challenging decision to make. I initially tried to avoid that as you said but the problem is `apollonEditor` is created and set in the same method that contains the only usage of `generateFeedbackFromAssessment`. You can check the relevant code [here](https://github.com/ls1intum/Artemis/blob/2fe16fc0b57622ec0ddbc9c90137dc7e7cba4e0d/src/main/webapp/app/exercises/modeling/assess/modeling-assessment.component.ts#L112-L126). So I couldn't write `jest.spyOn(comp.apollonEditor!, 'subscribeToSelectionChange')`. I tried adding a `spyOn` to setter of `apollonEditor` but since it is a basic field as opposed to a property, `jest.spyOn(comp!, 'apollonEditor', 'set')` didn't work either. And since the method definition seems like it has self-contained logic that is already documented, I think it is suitable to be a public method as well.

If you know a way that I can make `apollonEditor.subscribeToAssessmentChange` call `generateFeedbackFromAssessment` under these circumstances that would be very helpful.

Reply:
In this case, it is reasonable to me. 👍